### PR TITLE
Restore function 'PNE_EcrireJeuDeDonneesMPS_avecNom'

### LINKED ISSUE #4: Function SRSwritempsprob does not write MPS

### PR DESCRIPTION
**Pull request by flomnes**

### CODE REVIEW

**Comment by klorel** [comment]
Nothing to say on the code. From a design point of view, how to put the objoffset into the MPS ?

**Comment by flomnes** [comment]
> If a row is not mentioned anywhere in the RHS section, that row takes a right-hand side value of 0. You may define an objective offset by setting the negative offset as right-hand side of the objective row. For example, if the linear objective row in the problem is called COST and you want to add an offset of 1000 to your objective function, you can add the following to the RHS section :
```
RHS
    RHS1      COST      -1000
```

Sources :
https://www.ibm.com/docs/en/icos/20.1.0?topic=standard-records-in-mps-format
https://www.gurobi.com/documentation/9.5/refman/mps_format.html

**Comment by flomnes** [comment]
@klorel Merged the objective offset feature, included it in the MPS write.
Please note that `SPX_EcrireProblemeSpxAuFormatMPS`, `SPX_EcrireProblemeAuFormatMPS` and `SPX_EcrireJeuDeDonneesLineaireAuFormatMPS` are not called in the `SRS_PROBLEM` API.

File: src/SRS/srs_problem.c [hunk 1]
```diff
@@ -478,7 +480,7 @@ int SRSgetobjval(SRS_PROBLEM * problem_srs, double * objVal) {
 	if (problem_srs->maximize)
 		(*objVal) *= -1.;
 
-    *objVal += problem_srs->objective_offset;
```
Comment (klorel) [inline src/SRS/srs_problem.c:481]: may be this line should be beffor line 478

File: src/SRS/srs_problem.c [hunk 2]
```diff
@@ -478,7 +480,7 @@ int SRSgetobjval(SRS_PROBLEM * problem_srs, double * objVal) {
...
+    *objVal += problem_mps->objective_offset;
```
Comment (klorel) [inline src/SRS/srs_problem.c:483]: this line should be before line 480 to take into account the min/max

File: src/SRS/srs_problem.c [hunk 3]
```diff
@@ -734,13 +736,13 @@ int SRSchgrangeval(SRS_PROBLEM * problem_srs, size_t nbRowIndexes, const int * r
 
 int SRSsetobjoffset(SRS_PROBLEM* problem_srs, double value)
 {
-    problem_srs->objective_offset = value;
+    problem_srs->problem_mps->objective_offset = value;
```
Comment (klorel) [inline src/SRS/srs_problem.c:739]: should take into account the maximize attribute value

File: src/PNE/pne_ecrire_jeu_de_donnees_mps.c [hunk 4]
```diff
@@ -190,6 +184,12 @@ for ( Var = 0 ; Var < NombreDeVariables ; Var++ ) {
 
 /* RHS */
 fprintf(Flot,"RHS\n");
+// Objective offset
+// see https://www.ibm.com/docs/en/icos/20.1.0?topic=standard-records-in-mps-format
+// NOTE: By convention, we write here the negative objective value
+if (objective_offset != 0.0) {
+   fprintf(Flot,"    RHSVAL    OBJECTIF  %f\n",  -objective_offset);
```
Comment (klorel) [inline src/PNE/pne_ecrire_jeu_de_donnees_mps.c:191]: take into account the maximize attribute value

File: src/SRS/srs_problem.c (hunk 2 above)
Comment (flomnes) [reply src/SRS/srs_problem.c:483]: As discussed, not really
```
argmax {a.x + b} = argmax {a.x} + b = -argmin {-a.x} + b
```

**Pull request merged**